resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -78,7 +78,10 @@ private[spark] class KubernetesDriverConf(
 
   override val resourceNamePrefix: String = {
     val custom = if (Utils.isTesting) get(KUBERNETES_DRIVER_POD_NAME_PREFIX) else None
-    custom.getOrElse(KubernetesConf.getResourceNamePrefix(appName))
+    val resourceNamePrefix = custom.getOrElse(KubernetesConf.getResourceNamePrefix(appName))
+    // If the first character of resourceNamePrefix is number,add the extra prefix : "spark-".
+    val prefix = "spark-" + resourceNamePrefix.charAt(0)
+    resourceNamePrefix.replaceAll("^[0-9]", prefix)
```

Comment:
BTW, did you run the test locally?

Reply:
1. For example: resourceNamePrefix = 1min-xxx

After execute this code:

`val prefix = "spark-" + resourceNamePrefix.charAt(0)
    resourceNamePrefix.replaceAll("^[0-9]", prefix)`

resourceNamePrefix = spark-1min-xxx,which satisfies the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?`

2. For example : resourceNamePrefix = min-xxx

After execute this code:

`val prefix = "spark-" + resourceNamePrefix.charAt(0)
    resourceNamePrefix.replaceAll("^[0-9]", prefix)`

resourceNamePrefix = min-xxx,which does not change.





Local testing was done for this method only.